pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -162,6 +162,25 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	return nil
 }
 
+// Test runs the devfile test command
+func (a Adapter) Test(testcmd versionsCommon.DevfileCommand, show bool) (err error) {
+	componentExists := utils.ComponentExists(a.Client, a.ComponentName)
+	if !componentExists {
+		return fmt.Errorf("component does not exist, a valid component is required to run 'odo test'")
+	}
+	pod, err := a.waitAndGetComponentPod(true)
```

Comment:
Is there any reason to wait? Might cause performance issues. Instead maybe use https://github.com/openshift/odo/blob/b1d8462e7bce01e41e24e2e853d4203e32d91408/pkg/kclient/pods.go#L153

Reply:
The pod has to be started in order to run the test command. 

Actually, if you search in our code base, `GetPodUsingComponentName` is never being used. This might because of the same reason, it does not guarantee the pod is in a healthy state.

Reply:
> Actually, if you search in our code base, GetPodUsingComponentName is never being used

It has been used here https://github.com/openshift/odo/blob/e367c334c2e6a15f584821c34af4659bb2658a30/pkg/debug/portforward.go#L55 and a occlient version was used here https://github.com/openshift/odo/blob/e367c334c2e6a15f584821c34af4659bb2658a30/pkg/debug/portforward.go#L65.

> it does not guarantee the pod is in a healthy state.

@yangcao77 Even waitAndGetComponentPod() doesn't guarantee that. It might be in a `crashLoopBackOff` state and thus this command will have to wait till timeout and without spinners, it will feel unresponsive. Thus it's better to exit immediately for a better and faster user experience.

Reply:
updated